lib/lints/avoid_unnecessary_type_assertions/avoid_unnecessary_type_assertions_fix.dart

```diff
@@ -0,0 +1,54 @@
+part of 'avoid_unnecessary_type_assertions_rule.dart';
+
+/// A Quick fix for `avoid-unnecessary-type-assertions` rule
+/// Suggests to remove unnecessary assertions
+class _UnnecessaryTypeAssertionsFix extends DartFix {
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    AnalysisError analysisError,
+    List<AnalysisError> others,
+  ) {
+    context.registry.addIsExpression((node) {
+      // checks that the literal declaration is where our warning is located
+      if (!analysisError.sourceRange.intersects(node.sourceRange)) return;
+
+      final changeBuilder = reporter.createChangeBuilder(
+        message: "Remove unnecessary 'is'",
+        priority: 1,
+      );
+
+      final isOperatorOffset = node.isOperator.offset;
+      changeBuilder.addDartFileEdit((builder) {
+        builder.addDeletion(
+          SourceRange(
+            isOperatorOffset,
+            node.length - (isOperatorOffset - node.offset),
```

Comment:
Not sure we need the parentheses.
A better option might be to put the result of this calculation into a properly named variable.
```suggestion
            node.length - isOperatorOffset - node.offset,
```

Reply:
We need the parentheses here. But variables with proper names are better.

Reply:
Ah, right -- the grouping alters the sign of operation. So `a - (b - c)` -> `a + -(b + (-c))` -> `a + ((-b) - (-c))` -> `a + (- b + c)` -> `a - b + c`.

```suggestion
            node.length - isOperatorOffset + node.offset,
```

Reply:
BTW why is an integer variable named with a `isX` prefix?

Reply:
Because it is an offset of the 'is' operator. Maybe the better name is operatorIsOffset.

Reply:
`isOperatorIsOffset` -- sounds confusing
`operatorIsOffset` would make me double-take too.
Maybe `typeCheckOperatorOffset`?